keychain/derivation.go

```diff
@@ -20,6 +24,18 @@ const (
 	BIP0043Purpose = 1017
 )
 
+var (
+	// MaxKeyRangeScan is the maximum number of keys that we'll attempt to
+	// scan with if a caller knows the public key, but not the KeyLocator
+	// and wishes to derive a private key.
+	MaxKeyRangeScan = 100000
```

Comment:
i was curious about how long this would take so i tried it OMM, seems to take about 10s for an unknown key. dunno how long it would take on mobile, but seems like something we can try to optimize in the future to reduce recovery latency. not a blocker on the PR tho

Reply:
Yeah ideally this only gets thru a hundred or so keys before it finds the target key. This is meant to be a stop gap, as a future version of `lnd` can come with a database migration to start to store the full key locator for the sha chain root.